src/js/Draw/L.PM.Draw.Line.js

```diff
@@ -48,9 +48,14 @@ Draw.Line = Draw.extend({
         this._map.on('click', this._createVertex, this);
 
         // finish on double click
-        if(this.options.finishOnDoubleClick) {
+        if (this.options.finishOnDoubleClick) {
             this._map.on('dblclick', this._finishShape, this);
         }
+        // finish on layer event 
+        // #http://leafletjs.com/reference-1.2.0.html#interactive-layer-click
+        if (this.options.finishOn) {
```

Comment:
Would be cool to add a default to the options object so `click` is used when no `finishOn` option is specified

Reply:
would this be ok? `this._map.on(this.options.finishOn || 'dblclick', this._finishShape, this);`
I wouldn't put `click` since it would always clone polygon on first click :)

Reply:
the default should be inside the `options` object in the beginning of the class. It gets merged with the passed options, then you can just use `this.options.finishOn`.

Reply:
i see what you mean. but currently finishOnDoubleClick is turned off so I also added finishOn to null.